src/main/java/com/blackducksoftware/integration/jira/config/controller/BlackDuckConfigActions.java

```diff
@@ -0,0 +1,335 @@
+            config.setPasswordError("No Hub Password was found.");
+        }
+    }
+
+    private void validateProxyHostAndPort(final BlackDuckServerConfigSerializable config) {
+
+        if (StringUtils.isBlank(config.getHubProxyHost()) && StringUtils.isNotBlank(config.getHubProxyPort())) {
+            config.setHubProxyHostError("Proxy host not specified.");
+        }
+        if (StringUtils.isNotBlank(config.getHubProxyHost()) && StringUtils.isBlank(config.getHubProxyPort())) {
+            config.setHubProxyPortError("Proxy port not specified.");
+        } else if (StringUtils.isNotBlank(config.getHubProxyHost()) && StringUtils.isNotBlank(config.getHubProxyPort())) {
+            try {
+                final Integer timeout = Integer.valueOf(config.getHubProxyPort());
+                if (timeout <= 0) {
+                    config.setHubProxyPortError("Proxy port must be greater than 0.");
+                }
+            } catch (final NumberFormatException e) {
+                config.setHubProxyPortError(String.format("The String : %s, is not an Integer.", config.getHubProxyPort()));
+            }
+        }
+    }
+
+    public void validateProxyCredentials(final BlackDuckServerConfigSerializable config) {
+        if (StringUtils.isNotBlank(config.getHubProxyUser()) && StringUtils.isNotBlank(config.getHubProxyPassword()) && StringUtils.isBlank(config.getHubProxyHost())) {
+            config.setHubProxyHostError("Proxy host not specified.");
+        }
+        if (StringUtils.isNotBlank(config.getHubProxyUser()) && StringUtils.isBlank(config.getHubProxyPassword())) {
+            config.setHubProxyPasswordError("Proxy password not specified.");
+        } else if (StringUtils.isBlank(config.getHubProxyUser()) && StringUtils.isNotBlank(config.getHubProxyPassword())) {
+            config.setHubProxyUserError("Proxy user not specified.");
+        }
+
+    }
+
+    public void validateIgnoreHosts(final BlackDuckServerConfigSerializable config) {
+        final String ignoredProxyHosts = config.getHubNoProxyHosts();
+        if (StringUtils.isNotBlank(ignoredProxyHosts)) {
+            if (StringUtils.isBlank(config.getHubProxyHost())) {
+                config.setHubProxyHostError("Proxy host not specified.");
+            }
+            try {
+                if (ignoredProxyHosts.contains(",")) {
+                    String[] ignoreHosts = null;
+                    ignoreHosts = ignoredProxyHosts.split(",");
+                    for (final String ignoreHost : ignoreHosts) {
+                        Pattern.compile(ignoreHost.trim());
+                    }
+                } else {
+                    Pattern.compile(ignoredProxyHosts);
+                }
+            } catch (final PatternSyntaxException ex) {
+                config.setHubNoProxyHostsError("Proxy ignore hosts does not compile to a valid regular expression.");
+            }
+        }
+    }
+
```

Comment:
The methods below should be able to be private as well due to the same reason from above.